### ISSUE TITLE
Enhancement: failed calling webhook

### ISSUE THREAD

**Issue by vikas027** [senthilrch/kube-fledged#115]
I have used `helm` to install `kube-fledged` as below but unable to create `ImageCache`, not sure what I am missing.

```bash
❯ helm install kube-fledged kubefledged-charts/kube-fledged -n kube-fledged --create-namespace --version 0.8.2 --wait

❯ k get svc
NAME                          TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
kube-fledged-webhook-server   ClusterIP   172.20.98.94   <none>        3443/TCP   15m

❯ k get po                                                                                                            
NAME                                           READY   STATUS    RESTARTS   AGE
kube-fledged-controller-577fdf7fc7-gspfj       1/1     Running   0          15m
kube-fledged-webhook-server-687974b98c-v592g   1/1     Running   0          15m

❯ kubectl apply -f https://raw.githubusercontent.com/senthilrch/kube-fledged/master/deploy/kubefledged-imagecache.yaml                                                                                            
Error from server (InternalError): error when creating "https://raw.githubusercontent.com/senthilrch/kube-fledged/master/deploy/kubefledged-imagecache.yaml": Internal error occurred: failed calling webhook "validate-image-cache.kubefledged.io": Post "https://kube-fledged-webhook-server.kube-fledged.svc:3443/validate-image-cache?timeout=1s": context deadline exceeded
```

I have checked the logs but could not find what I could be missing.

```bash
❯ stern -l app.kubernetes.io/instance=kube-fledged
+ kube-fledged-controller-577fdf7fc7-gspfj › kube-fledged
+ kube-fledged-webhook-server-687974b98c-v592g › kube-fledged
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.754946       1 controller.go:123] Setting up event handlers
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.755701       1 main.go:76] Starting pre-flight checks
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.785219       1 controller.go:159] No dangling or stuck jobs found...
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.794035       1 controller.go:186] No dangling or stuck imagecaches found...
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.794050       1 main.go:80] Pre-flight checks completed
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.794063       1 controller.go:224] Starting kubefledged-controller
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.894754       1 controller.go:230] Informer caches synched successfull
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.894779       1 controller.go:236] Image cache worker started
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.894786       1 controller.go:240] Image cache refresh worker started
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.894801       1 image_manager.go:357] Starting image manager
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.894806       1 image_manager.go:360] Waiting for informer caches to sync
kube-fledged-controller-577fdf7fc7-gspfj kube-fledged I0917 19:16:39.995647       1 image_manager.go:365] Started image manager
kube-fledged-webhook-server-687974b98c-v592g kube-fledged I0917 19:16:37.599522       1 server.go:269] Wehook server listening on :443
```

**Comment by senthilrch** [comment]
@vikas027 : Thanks for raising this issue. The api-server is unable to establish a successful connection with kube-fledged's webhook server. The error you are seeing is not related to kube-fledged.

**Comment by vikas027** [comment]
> @vikas027 : Thanks for raising this issue. The api-server is unable to establish a successful connection with kube-fledged's webhook server. The error you are seeing is not related to kube-fledged.

Hey @senthilrch ,

This is a vanilla EKS cluster, everything else (like ArgoCD, Fluentbit,  Ambassador and a lot more) is working fine. 

It looks like others (https://github.com/senthilrch/kube-fledged/issues/123) are facing the same/similar issue too.

Can you please help me with some pointers what else we can do to troubleshoot the issue?

I did try calling the endpoint from other pods including `kube-fledged-controller`, didn't saw any issues

```
~ # curl -k "https://kube-fledged-webhook-server.kube-fledged.svc:3443/validate-image-cache?timeout=1s" -I
HTTP/2 200 
date: Sat, 27 Nov 2021 00:57:16 GMT

~ # 
```

**Comment by vikas027** [comment]
Hello @senthilrch , Did you get a chance to look at this? I have a hit dead end here :(

**Comment by senthilrch** [comment]
Hi @vikas027, I'll fix this in release v0.9.0

**Comment by vikas027** [comment]
Hello @senthilrch ,

Can we please reopen this issue?

I am still seeing the same issue on a vanilla EKS cluster. Also, I have a few questions

- It looks like we cannot create an `ImageCache` resource without the service `kube-fledged-webhook-server`. Is it true? If yes, why it is [disabled](https://github.com/senthilrch/kube-fledged/blob/master/deploy/kubefledged-operator/helm-charts/kubefledged/values.yaml#L10) by default?
- It looks like we can't [disable](https://github.com/senthilrch/kube-fledged/blob/master/deploy/kubefledged-operator/helm-charts/kubefledged/values.yaml#L60) `validatingWebhook` and [enable](https://github.com/senthilrch/kube-fledged/blob/master/deploy/kubefledged-operator/helm-charts/kubefledged/values.yaml#L10) `webhookServer`. If this is the case, shouldn't we have a common variable to enable/disable these?

I have tried a few combinations of the variables but can't see to make this work. My error is the same as what I have above.
